.github/workflows/doc-gen.yml

```diff
@@ -0,0 +1,44 @@
+name: Deploy static documentation on Pages
+
+on:
+  push:
```

Comment:
May be this workflow should only be launched if `/doc` folder has been modified. what do you think about?